app/src/lib/stats/stats-store.ts

```diff
@@ -481,6 +488,10 @@ export class StatsStore implements IStatsStore {
     ).length
     const diffMode = getShowSideBySideDiff() ? 'split' : 'unified'
 
+    const launchedFromApplicationsFolder = __DARWIN__
+      ? remote.app.isInApplicationsFolder()
+      : null
```

Comment:
I wondered if I should check `isInApplicationsFolder` too, like this:
```suggestion
    const launchedFromApplicationsFolder = __DARWIN__ && remote.app.isInApplicationsFolder
      ? remote.app.isInApplicationsFolder()
      : null
```

But decided not to do it because it feels "too defensive". Being on macOS and that method `undefined` should just not happen. Does that sound 👍 ?

Reply:
Also, is it ok to just import and use `remote.app` from here directly? I come from a codebase where we had dependency injection EVERYWHERE, so this "feels bad"… but, of course, it's very convenient 😂  Thoughts?

Reply:
> I wondered if I should check isInApplicationsFolder too, like this:

Eh, I'd approve either option but I'd add a comment so that future readers know why we're special casing. If you wanted to you could use optional chaining and nullish coalescing here:

```suggestion
    // isInApplicationsFolder is undefined when not running on Darwin
    const launchedFromApplicationsFolder =
      remote.app.isInApplicationsFolder?.() ?? null
```

> Also, is it ok to just import and use `remote.app` from here directly? I come from a codebase where we had dependency injection EVERYWHERE, so this "feels bad"… but, of course, it's very convenient 😂 Thoughts?

Yup! Totally okay! We don't use DI. We used to in the classic app but it got so abstract and caused so much boilerplate that we felt heavily constrained by it and  for the electron app we just haven't seen any compelling reason to (yet).

Reply:
> If you wanted to you could use optional chaining and nullish coalescing
Coalescing undefined to null? 🤯  What a time to be alive 😂 

> We used to in the classic app but it got so abstract and caused so much boilerplate that we felt heavily constrained by it
I totally know that feel 😐 🤗

Thanks for the feedback!